inc/TRestAxionSolarHiddenPhotonFlux.h

```diff
@@ -0,0 +1,133 @@
+/*************************************************************************
+ * This file is part of the REST software framework.                     *
+ *                                                                       *
+ * Copyright (C) 2016 GIFNA/TREX (University of Zaragoza)                *
+ * For more information see http://gifna.unizar.es/trex                  *
+ *                                                                       *
+ * REST is free software: you can redistribute it and/or modify          *
+ * it under the terms of the GNU General Public License as published by  *
+ * the Free Software Foundation, either version 3 of the License, or     *
+ * (at your option) any later version.                                   *
+ *                                                                       *
+ * REST is distributed in the hope that it will be useful,               *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          *
+ * GNU General Public License for more details.                          *
+ *                                                                       *
+ * You should have a copy of the GNU General Public License along with   *
+ * REST in $REST_PATH/LICENSE.                                           *
+ * If not, see http://www.gnu.org/licenses/.                             *
+ * For the list of contributors see $REST_PATH/CREDITS.                  *
+ *************************************************************************/
+
+#ifndef _TRestAxionSolarHiddenPhotonFlux
+#define _TRestAxionSolarHiddenPhotonFlux
+
+#include <TRestAxionSolarFlux.h>
+#include <TRestAxionSolarModel.h>
+
+//! A metadata class to load tabulated solar hidden photon fluxes. Mass and coupling set to 1.
+class TRestAxionSolarHiddenPhotonFlux : public TRestAxionSolarFlux {
+   private:
+    /// The hidden photon mass used for calculation
+    double HiddenPhotonMass;
```

Comment:
I think we do not need the mass here? It is the method `GetRandomEnergyAndRadius` where we define the mass. There should be something inside that method that gets the probability depending on the mass.

Reply:
Yes, this is possible.

We just need to make sure then that we have this mass inside `TRestAxionSolarFlux` and not within `TRestAxionQCDSolarFlux` or `TRestAxionHiddenPhotonFlux`.

So, we define a `TRestAxionSolarFlux::fMass`.

Then, inside the `TRestAxionGeneratorProcess` we can call `TRestAxionSolarFlux::SetMass(m)` before calling `GetRandomEnergyAndRadius`.

`TRestAxionGeneratorProcess` will only have access to those methods available to `TRestAxionSolarFlux`.

Reply:
Then, inside `TRestAxionSolarFlux` we should add:

```
Double_t fMass; //!

Double_t GetMass() { return fMass; }
void SetMass(const Double_t &m ){ fMass = m; }
```

Reply:
Ok, then would we remove the `double_t mass` from the other functions and just have them use `fMass`? For example in `GetTotalFlux(Double_t mass = 0)`